doc/langdef.md

```diff
@@ -116,44 +126,185 @@ In general it is a bad idea for those defining contexts or extensions to use
+interpreted as octet values rather than as Unicode code points. Both raw and
+multiline string literals can be used for byte literals.
+
+Escape sequences are a backslash (`\`) followed by one of the following:
+
+*   A single-quote(`'`), double-quote(`"`), or backslash(`\`), representing
+    itself.
+*   A code for whitespace:
+    *   `b`: backspace
+    *   `f`: form feed
+    *   `n`: line feed
+    *   `r`: carriage return
+    *   `t`: horizontal tab
+*   A `u` followed by four hexadecimal characters, encoding a Unicode code point
+    in the BMP. Characters in other Unicode planes can be represented with
+    surrogate pairs.
+*   Three octal digits, in the range `000` to `377`. For strings, it denotes the
+    unicode code point. For bytes, it represents an octet value.
+
+Examples:
+
+CEL Literal  | Meaning
+------------ | --------------------------------------------------------------
+`""`         | Empty string
+`'""'`       | String of two double-quote characters
+`'''x''x'''` | String of four characters "x''x"
+`"\""`       | String of one double-quote character
+`"\\"`       | String of one backslash character
+`r"\\"`      | String of two backslash characters
+`b"abc"`     | Byte sequence of 61, 62, 63
+`b"&yuml;"`  | Sequence of bytes 204 and 191 (UTF-8 of &yuml;)
+`b"\303\277"`| Also sequence of bytes 204 and 191
+`"\303\277"` | String of "&Igrave;&iquest;" (code points 204, 191)
+`"\377"`     | String of "&yuml;" (code point 255)
+`b"\377"`    | Sequence of byte 255 (_not_ UTF-8 of &yuml;)
+
+### Aggregate Values
+
+Lists are ordered sequences of values.
+
+Maps are a set of key values, and a mapping from these keys to arbitrary values.
+Key values must be an allowed key type: `int`, `uint`, `bool`, or `string`.
+
+Note that the type checker uses a finer-grained notion of list and map types.
+Lists are `list(A)` for the homogenous type `A` of list elements. Maps are
+`map(K, V)` for maps with keys of type `K` and values of type `V`.  The type
+`dyn` is used for heterogeneous values  See
+[Gradual Type Checking](#gradual-type-checking).
+
+Any protocol buffer message is a CEL value, and each message type is its own CEL
+type, represented as its name (e.g. `google.protobuf.timestamp`).
+Some message types will be declared to be _opaque_, which means that CEL
+expressions are not allowed to form literals of this type or to access the
+message fields with the selection operator (see below). Opaque messages can only
+be provided by variables in the evaluation context or returned from functions,
+and they can only be used by providing them as arguments to functions. For
+instance, messages of type `google.protobuf.timestamp` are opaque, and CEL
+expressions will not name their fields directly.
+
+A list can be denoted by the expression `[e1, e2, ..., eN]`, a map by `{ek1:
+ev1, ek2: ev2, ..., ekN: evN}`, and a message by `M{f1: e1, f2: e2, ..., fN: eN}`,
```

Comment:
It's subtle, but you may want to mention that in the message construction
case the field names are expected to be identifiers, and for maps the key
values may be an expression that resolves to `string`,`int`, `uint`, or
`bool` value

Reply:
Done.